system/gsettings.py

```diff
@@ -0,0 +1,258 @@
+        # commiting to get_value.
+        if schema not in schemas:
+            raise ValueError('{0} is not a valid schema'.format(schema))
+
+        if path is None:
+            settings = Gio.Settings(schema=schema, path=path)
+        else:
+            settings = Gio.Settings.new_with_path(schema_id=schema, path=path)
+
+        if key not in Gio.Settings.list_keys(settings):
+            raise ValueError('{0} is not valid in the schema: {1}'.format(key, schema))
+        return settings.get_value(key)
+
+    def set_value(self, schema, key, value):
+        """ Write a value to the specified key """
+        if ':' in schema:
+            schema, path = schema.split(':', 1)
+        else:
+            path = None
+
+        # No need to check values passed as they were
+        # already checked in the get_value function.
+        settings = Gio.Settings(schema=schema, path=path)
+        settings.set_value(key, value)
+
+    def cleanup(self):
+        """ Class cleanup """
+        if self._cleanup:
+            # Stop the dbus daemon if one was started.
+            time.sleep(1)
+            os.kill(int(self.session_pid), signal.SIGTERM)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            settings=dict(required=True, type='list')
+        ),
+        supports_check_mode=True,
+    )
+
+    has_changed = False
+    change_msgs = []
+
+    if not HAS_GIO:
+        module.fail_json(rc=1, msg='The python PyGObject module is a required dependency.')
+
+    # Prevent stack trace if the runtime directory is not writable.
+    work_dir = os.path.expanduser('~/.cache/dconf')
+    if not os.access(work_dir, os.W_OK):
+        module.fail_json(rc=1, msg='Unable to persist changes - please ensure {0} exists and is writable.'.format(work_dir))
+
+    gs = Gsettings()
+    gsettings = module.params['settings']
+    for gsetting in gsettings:
+        schema = gsetting['schema']
+        key = gsetting['key']
+        value = gsetting['value']
+        try:
+            old_setting = gs.get_value(schema, key)
+        except ValueError, err:
+            module.fail_json(rc=1, msg=err.message)
+
+        setting_type = old_setting.get_type_string()
+
+        old_value = old_setting.unpack()
+        if old_value != value:
+            has_changed = True
+            if module.check_mode:
+                change_msgs.append('{0}/{1} -> current: {2} ; proposed new: {3}'.format(schema, key, old_value, value))
+            else:
+                try:
+                    new_value = GLib.Variant(setting_type, value)
+                except Exception, err:
+                    module.fail_json(rc=1, msg='The value provided is invalid: {1}'.format(err))
+
+                gs.set_value(schema, key, new_value)
+                change_msgs.append('{0}/{1} -> old: {2} ; new: {3}'.format(schema, key, old_value, value))
+    gs.cleanup()
+    module.exit_json(rc=0, changed=has_changed, msg=change_msgs)
```

Comment:
You could remove the `has_changed` variable and use instead `bool(change_msgs)`.